src/Util/Printer.php

```diff
@@ -72,9 +72,19 @@ public function __construct($out = null)
      */
     public function flush()
     {
-        if ($this->out && \strncmp($this->outTarget, 'php://', 6) !== 0 && $this->out !== STDOUT && $this->out !== STDERR) {
-            \fclose($this->out);
+        if (!$this->out) {
+            return;
         }
+        if (\strncmp($this->outTarget, 'php://', 6) === 0) {
+            return;
+        }
+        if (defined('STDOUT') && $this->out === STDOUT) {
```

Comment:
`STDIN`, `STDOUT`, and `STDERR` are always defined for the PHP CLI SAPI. And PHPUnit is only supposed to be used with the CLI SAPI.

Reply:
> And PHPUnit is only supposed to be used with the CLI SAPI.

Oh. Some strange. How about difference between successful unit test on `cli` and fail tested code on `production` `fpm`? This is real case?

Reply:
Or you tell about another case?

Reply:
So... 
> STDIN, STDOUT, and STDERR are always defined for the PHP CLI SAPI.

what about undefined constant in this test?
```php
class CustomPrinter extends ResultPrinter
{
    public function __construct(
        $out = null,
        $verbose = false,
        $colors = self::COLOR_DEFAULT,
        $debug = false,
        $numberOfColumns = 80,
        $reverse = false
    ) {
        parent::__construct(STDOUT, $verbose, $colors, $debug, $numberOfColumns, $reverse);
    }

}
```
produced
```sh
+Warning: Use of undefined constant STDOUT - assumed 'STDOUT' (this will throw an Error in a future version of PHP) in /var/www/html/tests/_files/CustomPrinter.php on line 14

```

this is not "always"